pilot/pkg/model/network.go

```diff
@@ -472,21 +483,32 @@ func (n *networkGatewayNameCache) resolveFromCache(name string) []string {
 }
 
 func (n *networkGatewayNameCache) resolveAndCache(name string) []string {
-	if entry, ok := n.cache[name]; ok {
+	entry, ok := n.cache[name]
+	if ok {
 		entry.timer.Stop()
 	}
 	delete(n.cache, name)
-	addrs, ttl := n.resolve(name)
+	addrs, ttl, err := n.resolve(name)
 	// avoid excessive pushes due to small TTL
 	if ttl < MinGatewayTTL {
 		ttl = MinGatewayTTL
 	}
 	expiry := time.Now().Add(ttl)
+	refreshPeriod := ttl
+	if len(addrs) == 0 {
+		// no answer, retry after 30s (by default)
+		// ttl is set to ~136 years
+		refreshPeriod = DNSRetryPeriod
```

Comment:
For addrs=0, the DNS server can still return a valid TTL I think. If not, we would already default it to MinGatewayTTL which s the same as DNSRetryPeriod. So I think we should remove this, to allow the DNS server to set a TTL >30s when there are 0 IPs.

Reply:
> For addrs=0, the DNS server can still return a valid TTL I think

I don't think so, TTL is for each answer(address)

> If not, we would already default it to MinGatewayTTL which s the same as DNSRetryPeriod

If len(addrs)==0, the TTL will be 136y, why would it be `MinGatewayTTL`?

Reply:
A dns server can return a ttl for an empty response I believe

Reply:
Oh yes, sorry I was mixed up, let me fix this

Reply:
Please remove hold if it looks good to you🙂